axon/shaders/Makefile

```diff
@@ -1,8 +1,11 @@
 # Makefile for glslc compiling of HLSL files for compute
+# The go generate command does this automatically.
 
-all: gpu_cycle.spv
+all: 
+	cd ../; gosl -exclude=Update,UpdateParams,Defaults,AllParams github.com/goki/mat32/fastexp.go github.com/emer/etable/minmax ../chans/chans.go ../chans ../kinase ../fsfffb/inhib.go ../fsfffb github.com/emer/emergent/etime github.com/emer/emergent/ringidx rand.go avgmax.go neuromod.go context.go neuron.go synapse.go pool.go layervals.go act.go act_prjn.go inhib.go learn.go layertypes.go layerparams.go deep_layers.go rl_layers.go pvlv_layers.go pcore_layers.go prjntypes.go prjnparams.go deep_prjns.go rl_prjns.go pvlv_prjns.go pcore_prjns.go gpu_hlsl	
 
+# note: gosl automatically compiles the hlsl files using this command:
 %.spv : %.hlsl
-	glslc -fshader-stage=compute -o $@ $<
+	glslc -fshader-stage=compute -O --target-env=vulkan1.1 -o $@ $<
```

Comment:
This generates the spv files, right? Can we add `*.spv` to gitignore then?

Reply:
actually the `*.spv` are added to git so that anyone can just checkout and build without requiring the glslc stuff.  Only need to regenerate when modifying the code, and they are universal.

Reply:
I think that's a bad idea
1. Will diverge if people who don't have Macs update the Go code. Will confuse people that try to update the go code but don't have glslc installed. How do we remember to regenerate them?
2. glslc is actually pretty easy to install.
3. Pollutes git. I don't think git stores diffs for binaries, and instead stores the whole binary at every commit. This gets too large over time.

Reply:
Maybe @garymmm can weight in

Reply:
If we don't check in the .spv files, can `go build` work (assuming glslc is installed)?
If not, what is the workflow?

For 3, I agree that having many revisions of binary blobs may lead to the repo getting annoyingly large pretty quickly.
We could use [git LFS](https://docs.github.com/en/repositories/working-with-files/managing-large-files/configuring-git-large-file-storage), but I'm not sure how much better that makes it.

Can we make it optional to build the GPU code using build tags? If so then I think that'd be ideal. Then people who want to use the GPU code can install all glslc, and people who don't are not burdened by having all these files in their checkout and the slower build and link times.

Reply:
@siboehm the .spv files are NOT platform specific.  you get the same .spv file on any platform. it is like a virtual bytecode representation of the program, which is then compiled into the actual machine language on loading.

the files are between 2k and 60k in size, roughly 20k average -- there are 21 of them -- so roughly 420k total.  most will not change because the code they involve is unlikely to change, and the .spv captures only the relevant bits of code.

maybe watch it a bit and see how it goes in terms of git churn?

Reply:
I'm not concerned about platform specificity. Python Bytecode isn't platforms specific either, and yet never goes into git for the same 3 reasons I listed.

Reply:
python bytecode is auto-generated and nobody ever worries about it.  the go generate step and glslc compiler install are extra steps for anyone who just wants to "go build".  

Also note that anyone who can run the gui can use the GPU mode (i.e., anyone using the software at all), so I don't think it makes sense to make this a conditional part of the build.  With the data parallel thing in place, it will likely be a win even for small size models, so we'll likely start turning gpu on by default in many cases.

Reply:
THe issue of divergence between the generated and source-of-truth remains, especially without automated testing. But if you volunteer to fix things when they get out of sync, I guess this is OK for now.

Reply:
Anyone pushing a PR should do `go generate` to update the .spv files as needed, and run the GPU test on their machine.  Ideally we'll figure out how to get testing working with the software vulkan thing too.  But in any case, I can always fix stuff as issues arise.